packages/deployments/contracts/contracts/core/connext/libraries/LibCrossDomainProperty.sol

```diff
@@ -3,158 +3,109 @@ pragma solidity 0.8.15;
-   * @param _property The property
-   * @return True if the property is of type DomainAndSender
-   */
-  function isDomainAndSender(bytes29 _property) internal pure returns (bool) {
-    return isValidPropertyLength(_property) && isType(_property, Types.DomainAndSender);
-  }
-
-  /**
-   * @notice Retrieves the identifier from property
-   * @param _property The property
-   * @return The property type
-   */
-  function propertyType(bytes29 _property) internal pure returns (uint8) {
-    return uint8(_property.indexUint(0, 1));
-  }
-
-  /**
-   * @notice Converts to a Property
-   * @param _view The property
-   * @return The newly typed property
-   */
-  function tryAsProperty(bytes29 _view) internal pure returns (bytes29) {
-    if (isValidPropertyLength(_view)) {
-      return _view.castTo(uint40(Types.DomainAndSender));
-    }
-    return TypedMemView.nullView();
-  }
-
-  /**
-   * @notice Asserts that the property is of type DomainAndSender
-   * @param _view The property
-   * @return The property
-   */
-  function mustBeProperty(bytes29 _view) internal pure returns (bytes29) {
-    return tryAsProperty(_view).assertValid();
-  }
-
-  /**
-   * @notice Retrieves the sender from a property
-   * @param _property The property
-   * @return The sender address
+   * @notice Appends the supplied properties to the end of the callData
+   * @dev See warning at the top of the file about usage
+   * @param _amount The amount the call is executed with
+   * @param _originDomain The originating domain of the call
+   * @param _originSender The msg.sender of the xcall
+   * @param _calldata The function data to be executed with the .call
+   * @return bytes A byte array containing the calldata and all properties
    */
-  function sender(bytes29 _property) internal pure typeAssert(_property, Types.DomainAndSender) returns (address) {
-    // before = 1 byte id + 4 bytes domain = 5 bytes
-    return _property.indexAddress(5);
+  function formatCalldataWithProperties(
+    uint256 _amount,
+    uint32 _originDomain,
+    address _originSender,
+    bytes memory _calldata
+  ) internal returns (bytes memory) {
+    // Convert data to bytes
+    bytes memory properties = abi.encodePacked(_amount, _originDomain, _originSender);
+    return bytes.concat(_calldata, properties);
   }
 
   /**
-   * @notice Retrieves the domain from a property
-   * @param _property The property
-   * @return The sender address
+   * @notice Parses the amount from the msg.data
+   * @dev See warning at the top of the file about usage
+   * @param _data The msg.data sent by executor
+   * @return uint256 Amount to execute call with
    */
-  function domain(bytes29 _property) internal pure typeAssert(_property, Types.DomainAndSender) returns (uint32) {
-    // before = 1 byte identifier = 1 byte
-    return uint32(_property.indexUint(1, 4));
+  function amount(bytes memory _data) internal returns (uint256) {
```

Comment:
Just confirming that the `_data` passed here does not include the concatenated calldata but just the properties component as seen in `formatCalldataWithProperties()`.

Reply:
@0xleastwood I think _data is the full `msg.data`, via `callDataLength()` the orignal calldata is skipped.

Reply:
yep! full `msg.data`